vmm/src/vm.rs

```diff
@@ -1683,6 +1685,42 @@ impl Vm {
             }
         }
     }
+
+    /// Gets the total actual size of the RAM.
+    pub fn get_zones_actual_size(&self) -> Result<Option<Vec<VmZoneActualSize>>> {
+        let memory_config = &self.config.lock().unwrap().memory;
+
+        if memory_config.hotplug_method == HotplugMethod::VirtioMem {
+            let mut zones_actual_size = Vec::new();
+            if let Some(zones) = &memory_config.zones {
+                for zone in zones.iter() {
+                    zones_actual_size.push(VmZoneActualSize {
+                        id: zone.id.clone(),
+                        size: self
+                            .memory_manager
+                            .lock()
+                            .unwrap()
+                            .get_zone_actual(&zone.id)
+                            .map_err(Error::GetZoneActual)?,
+                    });
+                }
+            } else {
+                zones_actual_size.push(VmZoneActualSize {
+                    id: DEFAULT_MEMORY_ZONE.to_string(),
+                    size: self
+                        .memory_manager
+                        .lock()
+                        .unwrap()
+                        .get_zone_actual(DEFAULT_MEMORY_ZONE)
+                        .map_err(Error::GetZoneActual)?,
+                });
+            }
+
+            Ok(Some(zones_actual_size))
+        } else {
+            Ok(None)
+        }
+    }
```

Comment:
Could you please factorize the code a bit? The function `get_total_ram_actual()` is almost similar, and it is also needed for `vm.info`, therefore I think you could have one big function returning all you need for vm.info to print what you need. At the moment, there's too much duplication IMO.

Reply:
Updated.